### ISSUE TITLE
PLUGIN - local_auth returning empty array for .getAvailableBiometrics() for iOS Simulator device

### ISSUE THREAD

**Issue by sjmcdowall** [flutter/flutter#25635]
I upgraded to the latest `local_auth` and tried out the new routines `.canCheckBiometrics()` and then if `true`, `.getAvailableBiometrics()`.

However, when I run on an iOS iPhone XR Simulator (for example) I am getting an empty array for the `.getAvailableBiometrics()` but the `.canCheckBiometrics()` returns `true` (as one would expect).

Here is the code (snippet):
```dart
else if (Platform.isIOS) {
        deviceData = _readIosDeviceInfo(await deviceInfoPlugin.iosInfo);
        _setiOSDeviceInfo();

        // Do the Bio check
        hasBiometrics = await localAuth.canCheckBiometrics;
        print('[Device - init() IOS] - hasBiometrics is $hasBiometrics');
        if (hasBiometrics) {
          // iOS is a bit more complex since it can be either Touch or Face ID
          List<BiometricType> availableBiometrics =
              await localAuth.getAvailableBiometrics();

          print('[Device - init() IOS] - Available Biometrics are:');
          print(availableBiometrics);
          if (availableBiometrics.contains(BiometricType.face)) {
            print('[Device - init() IOS] - Found FaceID biometrics');
            biometricType = 'Face ID';
          } else if (availableBiometrics.contains(BiometricType.fingerprint)) {
            print('[Device - init() IOS] - Found TouchID biometrics');
            biometricType = 'Touch ID';
          } else {
            print('[Device - init() IOS] - Didn' 't find any biometrics??');
            biometricType = 'Unknown';
          }
        }
      }
```

And the result of run (i.e. the debug) -- 

```
flutter: [Device - setiOS()] - machineName is iPhone XR
flutter: [Device - setiOS()] - machineType is x86_64
flutter: [Device - setiOS()] - SIM platform is iPhone10
    1000 = 1;
}, (null) on <LAClient: 0x600003e96490>
flutter: [Device - init() IOS] - hasBiometrics is true
    1000 = 1;
}, (null) on <LAClient: 0x600003e93160>
flutter: [Device - init() IOS] - Available Biometrics are:
flutter: []
flutter: [Device - init() IOS] - Didnt find any biometrics??
```

This works on Android just fine -- and real iOS devices connected to USB (or via TestFlight etc).

Am I missing something?

**Comment by diegoveloper** [comment]
Go to your iPhone XR Simulator, select 'Hardware' at the top menu , next 'Face ID' and 'Enrolled' , after this you will have face id on your iOS simulator and you can use these options:

-  Hardware -> Face ID -> Matching Face
-  Hardware -> Face ID -> Non Matching Face

**Comment by sjmcdowall** [comment]
@diegoveloper -- Thank you.  I've been programming Flutter for 6 months and am quite familiar with how to work with the iOS simulator w/ Face / Touch ID.  

My point is that this routine, if at all possible, should return all possible biometrics on the device if "canCheckBiometrics()" is true IMHO.   Right now, you do need to Enroll in order to see that in the list.  This may certainly be a limitation on the iOS simulator or that my expectations are simply wrong .. 

Cheers!

**Comment by overgaardmorten** [comment]
Facing the same issues. This happens on IOS 9 & 10 because of the below code found in LocalAuthPlugin.m

```
if (@available(iOS 11.0.1, *)) {
        if (context.biometryType == LABiometryTypeFaceID) {
          [biometrics addObject:@"face"];
        } else if (context.biometryType == LABiometryTypeTouchID) {
          [biometrics addObject:@"fingerprint"];
        }
      }
``` 
When running ios9 & 10 nothing is added to *biometrics --> Empty array is returned to Dart..

;-)
/Morten

**Comment by sjmcdowall** [comment]
I’m on vacation with little internet but I think a PR for this is already submitted and waiting ? 

I have a work around til that is published but I can’t access the code right now - will update this once I can - although it’s pretty simplistic :)

**Comment by zoechi** [comment]
#27366 contains a running minimal reproduction

**Comment by kangwang1988** [comment]
Closed #25635 via https://github.com/flutter/plugins/pull/1173
